src/games/strategy/triplea/ui/RouteOptimizer.java

```diff
@@ -22,60 +22,115 @@
   private static final int commonAdditionalScreens = 2;
   private static final int minAdditionalScreens = 0;
 
-  private final double gameMapWidth;
+  private Point endPoint;
   private int mapWidth;
   private int mapHeight;
 
   public RouteOptimizer(MapData mapData, MapPanel mapPanel) {
     checkNotNull(mapData);
     this.mapPanel = checkNotNull(mapPanel);
-    mapWidth = mapPanel.getImageWidth();
-    mapHeight = mapPanel.getImageHeight();
-
     isInfiniteY = mapData.scrollWrapY();
-
     isInfiniteX = mapData.scrollWrapX();
-
-    gameMapWidth = mapData.getMapDimensions().getWidth();
-
   }
 
-  List<Point> normalizeForHorizontalWrapping(Point... route) {
-    final boolean crossesMapSeam = crossMapSeam(route);
-    final List<Point> points = Arrays.asList(route);
-
-    if (crossesMapSeam) {
-      final int minX = minX(route);
-      for (Point p : points) {
-        if ((p.x - minX) > mapPanel.getWidth()) {
-          p.x -= this.gameMapWidth;
+  /**
+   * Algorithm for finding the shortest path for the given Route
+   * 
+   * @param route The joints on the Map
+   * @return A Point array which goes through Map Borders if necessary
+   */
+  public Point[] getTranslatedRoute(Point... route) {
+    mapWidth = mapPanel.getImageWidth();
+    mapHeight = mapPanel.getImageHeight();
+    if (route == null || route.length == 0) {
+      // Or the array is too small
 
-        }
+      return route;
+    }
+    if (!isInfiniteX && !isInfiniteY) {
+      // If the Map is not infinite scrolling, we can safely return the given Points
+      endPoint = route[route.length - 1];
+      return route;
+    }
+    List<Point> result = new ArrayList<>();
+    Point previousPoint = null;
+    for (Point point : route) {
+      if (previousPoint == null) {
+        previousPoint = point;
+        result.add(point);
+        continue;
       }
+      Point closestPoint = getClosestPoint(previousPoint, getPossiblePoints(point));
+      result.add(closestPoint);
+      previousPoint = closestPoint;
     }
-    return points;
-  }
-
-  private boolean crossMapSeam(Point... route) {
-    final int minX = minX(route);
-    return Arrays.stream(route).filter(point -> (point.x - minX) > mapPanel.getWidth()).findAny().isPresent();
+    endPoint = result.get(result.size() - 1);
+    return result.toArray(new Point[result.size()]);
   }
 
-  private static int minX(Point... route) {
-    int minX = Integer.MAX_VALUE;
-    for (Point p : Arrays.asList(route)) {
-      if (p.x < minX) {
-        minX = p.x;
+  /**
+   * Returns the Closest Point out of the given Pool
+   * 
+   * @param source the reference Point
+   * @param pool Point2D List with all possible options
+   * @return the closest point in the Pool to the source
+   */
+  private Point getClosestPoint(Point source, List<Point2D> pool) {
```

Comment:
Technically I was recommending extraction, regardless.. Sure, those look like reasonable enough test cases.  I will admit that it does always seem fishy when a method has to be elevated in scope to test it. If also the logic performed in the method is also not terribly important to mock, then yeah, not extracting it is fine  : )